tests/simulate/test_dilute.py

```diff
@@ -0,0 +1,152 @@
+"""Unit tests for dilute module."""
+import tempfile
+import unittest
+
+import numpy as np
+
+import relentless
+
+class test_Dilute(unittest.TestCase):
+    """Unit tests for relentless.simulate.Dilute"""
+
+    #mock functions for use as operations
+    class UpdateEnsemble(relentless.simulate.SimulationOperation):
+        def __call__(self, sim):
+            try:
+                const = sim.constant_ens
+            except AttributeError:
+                const = False
+
+            if not const:
+                sim.ensemble.T += 2.0
+                sim.ensemble.V = relentless.Cube(L=4.0)
+                for t in sim.ensemble.types:
+                    sim.ensemble.N[t] += 1.0
+
+    class UpdatePotential(relentless.simulate.SimulationOperation):
+        def __call__(self, sim):
+            try:
+                const = sim.constant_pot
+            except AttributeError:
+                const = False
+
+            if not const:
+                for pair in sim.potentials:
+                    sim.potentials[pair]['r'] *= 2.0
+                    sim.potentials[pair]['u'] *= 3.0
+
+    def setUp(self):
+        self._tmp = tempfile.TemporaryDirectory()
+        self.directory = relentless.Directory(self._tmp.name)
+
+    def test_init(self):
+        """Test creation from data."""
+        operations = [self.UpdateEnsemble(), self.UpdatePotential()]
+        options = {'constant_ens':True, 'constant_pot':True}
+
+        #no operations, no options
+        d = relentless.simulate.Dilute()
+        self.assertCountEqual(d.operations, [])
+        self.assertDictEqual(d.options, {})
+
+        #with operations, no options
+        d = relentless.simulate.Dilute(operations)
+        self.assertCountEqual(d.operations, operations)
+        self.assertDictEqual(d.options, {})
+
+        #no operations, with options
+        d = relentless.simulate.Dilute(**options)
+        self.assertCountEqual(d.operations, [])
+        self.assertDictEqual(d.options, options)
+
+        #with operations, with options
+        d = relentless.simulate.Dilute(operations, **options)
+        self.assertCountEqual(d.operations, operations)
+        self.assertDictEqual(d.options, options)
```

Comment:
Most of these tests can just go in `test_simulate.py`. Maybe just do one to make sure you can setup a Dilute simulation with one or two of the `dilute` operations.